imageroot/actions/configure-module/10start_service

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+#
+# Copyright (C) 2022 Nethesis S.r.l.
+# http://www.nethesis.it - nethserver@nethesis.it
+#
+# This script is part of NethServer.
+#
+# NethServer is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or any later version.
+#
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+# Redirect any output to the journal (stderr)
+exec 1>&2
+
+systemctl --user start webtop
+
+timeout --foreground 10s bash -c "until podman container exists postgres ; do sleep 0.5; done"
```

Comment:
I didn't inspect postgres container, but usually it's not true.

Reply:
I don't understand. It's not a Postgres issue only: do you think `podman run --detach` completes before the container is created?

Reply:
I do not know, but probably yes if it must pull the image.
Still, event if the container is created, it doesn't mean it's really started: it could be blocked somewhere or fail as soon as podman returns.

Reply:
The postgres service is not started directly but is requested as dependency by the `webtop`service` and you cannot start the webtop container without first creating the `webtop` pod, created by the `webtop` service.

Reply:
Thank you Matteo, IIUC the original issue, the pod unit is "started" but individual containers still not. I didn't notice it!

So, if possible, I'd prefer to include the  "startup completed check" logic in the Systemd units that "supervise" the services. For instance,

1. in line 25 start postgres.service instead of webtop.service
1. implement the next wait of loop (line 32) in the postgres.service unit, as ExecStartPost.
1. ensure postgres.service starts promptly the webtop.service unit too (e.g. Requires=webtop.service, After=webtop.service)

If not possible, this first check could be entirely omitted because on the current line 32 the fact that `postgres` container exists is implicit.

Reply:
> 3. ensure postgres.service starts promptly the webtop.service unit too (e.g. Requires=webtop.service, After=webtop.service)

I will refactor the code in this way

> If not possible, this first check (and the next one too) could be entirely omitted because on the current line 32 the fact that postgres container and webtop5 DB exist is implicit.

Is not possible to determine implicit the status of the `webtop5` db, due to the initialization process of the postgres container. See: https://github.com/docker-library/postgres/issues/146#issuecomment-215856076